File: 0011-container-with-most-water/0011-container-with-most-water.java
import java.util.*;

class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length - 1;
        int result = 0;

        while (l < r) {
            int minHeight = Math.min(height[l], height[r]);
            int area = minHeight * (r - l);
            result = Math.max(result, area);

            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }

        return result;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r = len(height)-1
        result=0
        while l<r:
            min_height = min(height[l],height[r])
            area = min_height * (r-l)
            result = max(area,result)
            if height[l]<=height[r]:
                l+=1
            else:
                r-=1
        return result

